go/src/url_shortener/url_shortener.go

```diff
@@ -114,8 +123,52 @@ func (c *URLShortener) shortenHandler(w http.ResponseWriter, r *http.Request) {
 	c.incrementHandlerCounter(c.shortenRoute, true)
 }
 
+type redirectsJSON struct {
+	Success int `json:"success"`
+	Failed  int `json:"failed"`
+}
+
+type handlerJSON struct {
+	Name  string `json:"name"`
+	Count int    `json:"count"`
+}
+
+type serverStatsJSON struct {
+	TotalURL  int           `json:"total_url"`
+	Redirects redirectsJSON `json:"redirects"`
+	Handlers  []handlerJSON `json:"handlers"`
+}
+
+type statsJSON struct {
+	ServerStats serverStatsJSON `json:"server_stats"`
+}
+
 func (c *URLShortener) statisticsHandler(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintf(w, "%s", c.getStatistics())
+	url := r.URL
+	query := url.Query()
+	format := query["format"]
+
+	statsJSON := statsJSON{}
+
+	c.computeStatistics(&statsJSON)
```

Comment:
In Go it's preferable to to have computeStatistics directly returns a Statistics structure (pointer or not).
Even if it's a pointer, the compiler is clever enough to allocate statsJSON on the stack, since it sees that its lifetime ends with the scope of `statisticsHandler`.

The resulting code would IMHO be easier to use, just call:
```go
stats := c.computeStatistics()
```